tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,448 @@
-        createOffsetsTopic(listenerName(), new Properties());
-        String missingGroup = "missing.group";
+    @ClusterTest
+    public void testDeleteWithMixOfSuccessAndError() throws Exception {
+        try (ConsumerGroupExecutor executor = buildConsumerGroupExecutor(GROUP)) {
+            String missingGroup = "missing.group";
+            String[] cgcArgs = new String[]{"--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", GROUP};
+            ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+
+            TestUtils.waitForCondition(
+                    () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state, "Stable"),
+                    "The group did not initialize as expected.");
+
+            executor.close();
+
+            TestUtils.waitForCondition(
+                    () -> Objects.equals(service.collectGroupState(GROUP).state, "Empty"),
+                    "The group did not become empty as expected.");
+
+            cgcArgs = new String[]{"--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", GROUP, "--group", missingGroup};
+
+            ConsumerGroupCommand.ConsumerGroupService service2 = getConsumerGroupService(cgcArgs);
+            Map<String, Throwable> result = service2.deleteGroups();
+            assertTrue(result.size() == 2 &&
+                            result.containsKey(GROUP) && result.get(GROUP) == null &&
+                            result.containsKey(missingGroup) &&
+                            result.get(missingGroup).getMessage().contains(Errors.GROUP_ID_NOT_FOUND.message()),
+                    "The consumer group deletion did not work as expected");
+        }
+    }
 
-        // run one consumer in the group
-        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1);
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+    @ClusterTest
+    public void testDeleteWithUnrecognizedNewConsumerOption() {
+        String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", GROUP};
+        assertThrows(OptionException.class, () -> getConsumerGroupService(cgcArgs));
+    }
 
-        TestUtils.waitForCondition(
-            () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state, "Stable"),
-            "The group did not initialize as expected.");
+    private ConsumerGroupExecutor buildConsumerGroupExecutor(String group) {
+        return new ConsumerGroupExecutor(cluster.bootstrapServers(),
+                1,
+                null != group ? group : GROUP,
+                GroupProtocol.CLASSIC.name,
```

Comment:
The consumer group command supports deleting CONSUMER groups, not only CLASSIC ones, so shouldn't we parametrize this test to make sure checks deleting both group types? 

Not sure if I'm missing something or that's being tracked in a different jira maybe @dajac? (I can't find it)

Reply:
agree to add test for `CONSUMER`. The origin test case does not include that, and we can complete that in this PR. 

@frankvicky It seems to me the following changes can be added.

1. add `@ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true")`
2. use `cluster.isKRaftTest()` to avoid using `GroupProtocol.CONSUMER` in zk mode

Reply:
we will also need `@ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "consumer,classic")`  (along with `group.coordinator.new.enable=true `)

Reply:
> we will also need @ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "consumer,classic") (along with group.coordinator.new.enable=true )

It seems all we need is to set one of them. see https://github.com/apache/kafka/blob/1b301b30207ed8fca9f0aea5cf940b0353a1abca/core/src/main/scala/kafka/server/KafkaConfig.scala#L1170